### PR TITLE
Dev docs for banners

### PR DESCRIPTION
**Pull request by rami-a**

### CODE REVIEW

File: docs/components/dartpad/banners/regular/main.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      debugShowCheckedModeBanner: false,
+      home: MyHomePage(),
+      theme: ThemeData(
+        colorScheme: ColorScheme.light(),
+        primaryColor: ColorScheme.light().primary,
+        primaryColorDark: ColorScheme.light().primary,
```
Comment (perclasson) [inline docs/components/dartpad/banners/regular/main.dart:15]: This makes the banner purple? If so LGTM.

Comment (rami-a) [reply docs/components/dartpad/banners/regular/main.dart:15]: Yes it does

File: docs/components/banners.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+path: /catalog/material-banners/
+-->
+
+# Banners
+
+A [banner](https://material.io/components/banners) displays a prominent message and related optional actions.
+
+![Hero image of a banner showing a transaction error message](assets/banners/banners_hero.png)
+
+## Contents
```
Comment (guidezpl) [inline docs/components/banners.md:16]: nit: there should only be links to 2nd level headers, and there should be a link to #contents at the top

Comment (rami-a) [reply docs/components/banners.md:16]: gotcha, done

File: docs/components/banners.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+-------------- | ------------------------ 
+**Buttons** | `actions`
+
+
+## Banners example
+
+`MaterialBanner`
+- [Class definition](https://master-api.flutter.dev/flutter/material/MaterialBanner-class.html)
+- [GitHub source](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/banner.dart)
+- [Dartpad Demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/banners/regular)
```
Comment (mingjane-work) [inline docs/components/banners.md:91]: FYI, Google style guide uses sentence case for titles ("Dartpad demo")

Comment (rami-a) [reply docs/components/banners.md]: ah good to know, fixed. Thanks

**Pull request merged**